.github/workflows/release_notes.yml

```diff
@@ -0,0 +1,69 @@
+name: Generate Release Notes
+
+on:
+  pull_request:
+
+# on:
+#   push:
+#     branches:
+#       - main
+
+jobs:
+  send-pull-requests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Generate release notes and environment variables
+        run: |
+          ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file
+          echo "file_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.file_name)" >> $GITHUB_ENV
+          echo "branch_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.branch_name)" >> $GITHUB_ENV
+
+      - name: Build source file
```

Comment:
This step is separate from the above since environment variables can only be used outside the step in which they are created.

Reply:
I haven't fully understood why, but yes. Trying to build `source_file` directly using `${{env.file_name}}` makes changes in a GitHub workspace. We were only able to get this to build correctly by first adding the extra step to build the correct file name

Reply:
Gotcha. So my follow up question is... why do you need to create the file in the other step in the first place? Could you just move the file creation and name fetching to this section? That way you wouldn't need to create a new env var that doesn't do anything but reference the other step's env var.

Reply:
Resolved synchronously with James. These steps are now combined. 42a4a2b10ba783a592233a272a0c29266c639964